backend/GameManager.py

```diff
@@ -58,5 +82,173 @@ def get_pieces_left(self):
         }
 
     def end_turn(self):
-        self.turn = Turn.swap_turn(self.turn)
+        self.turn = Color.swap_turn(self.turn)
         return 1
+
+    def is_adjacent(self, current_row, current_col, target_row, target_col):
+        """ Check if the target position is adjacent to the current position """
+        return (target_row, target_col) in self.board.adjacent_positions_map.get((current_row, current_col), [])
+
+    def is_empty(self, row, col):
+        """ Check if the cell at (row, col) is empty """
+        return self.board.board[row][col] is None
+
+    def is_adjacent_and_empty(self, current_row, current_col, target_row, target_col):
```

Comment:
Feels like we dont need this function and can reuse a combination of `is_empty` and `is_adjacent`